src/coreclr/jit/optimizer.cpp

```diff
@@ -4858,7 +4862,7 @@ bool Compiler::optIfConvert(BasicBlock* block)
 
     // Create a select node.
     GenTreeConditional* select =
-        gtNewConditionalNode(GT_SELECT, cond, asgNode->gtGetOp2(), falseInput, asgNode->TypeGet());
+        gtNewConditionalNode(GT_SELECT, cond, asgNode->gtGetOp2(), falseInput, genActualType(destination));
```

Comment:
cc @a74nh, small bug fix here -- if the assignment is going to a small typed local it may have a small type, but this is not actually an allowed type on `GT_SELECT`. Instead this uses the type of the local itself (or `TYP_INT` if small), which should be compatible in the cases there is an implicit conversion (e.g. `TYP_I_IMPL <-> TYP_BYREF`).